sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -328,7 +328,8 @@ object StructType extends AbstractDataType {
   def apply(fields: Seq[StructField]): StructType = StructType(fields.toArray)
 
   def apply(fields: java.util.List[StructField]): StructType = {
-    StructType(fields.toArray.asInstanceOf[Array[StructField]])
+    import scala.collection.JavaConverters._
+    StructType(fields.asScala)
```

Comment:
Without this change, was there any issue when we use `org.apache.spark.sql.types.DataTypes.createStructType`? Or, the problem only appear when we use this apply method directly?